DOCSP-41323 made behavior page (#356)
mongodb/docs-cluster-to-cluster-sync: source/reference/mongosync-behavior.txt

.. _c2c-mongosync-behavior:

======================
``mongosync`` Behavior
======================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 3
   :class: singlecol

> Description

Reviewer: I think we could probably take out this "description" heading as well

{+c2c-product-name+}. ``mongosync`` migrates data from one cluster to
another and can keep the clusters in continuous sync. 

> For information on how the ``mongosync`` process works, see 
> :ref:`about-mongosync`. 
> 

Reviewer: [suggestion] for minimalism

```suggestion
For information on the ``mongosync`` synchronization process, see 
:ref:`about-mongosync`. 

```

> information on how the ``mongosync`` process works, see :ref:`about-mongosync`. 

Reviewer: [question] Do you think we can cut this line out because the purpose of the page is indicated by the title?

Author: I put this line in because I wanted to differentiate this page from the "About mongosync" page, but you are right that the title makes it redundant. I will take it out for now but let me know if you think there is a better way of summarizing this page is besides using the word "behavior" again.

> To get started with ``mongosync``, refer to the :ref:`Quick Start Guide
> <c2c-quickstart>`. For more detailed information, refer to the
> :ref:`c2c-install` or :ref:`c2c-connecting` page that best fits your
> situation. 

Reviewer: [suggestion] I would suggest splitting this up to be consistent with the space between the first more information line (that links to about mongosync) and these lines

```suggestion
To get started with ``mongosync``, refer to the :ref:`Quick Start Guide
<c2c-quickstart>`. 

For more detailed information, refer to the
:ref:`c2c-install` or :ref:`c2c-connecting` page that best fits your
situation. 
```

Author: I like this, fixing now

> .. note::

Reviewer: [question] Do you think it makes sense to keep this note here or is there another place/page that this might fit better?

Author: Yes, I do think it makes sense for this note to be here. I think it summarizes important facts that are scattered in different places, and I think that users will look to the behavior page for this type of information. There is the limitations page, but we link to it in this note and I think it's important for these limitations to be at the top of the behavior page. What are your thoughts?

Reviewer: That sounds like a fair reason to me!

   - For a full list of limitations, see :ref:`Cluster-to-Cluster Sync
     Limitations<c2c-limitations>`. 

> Behavior

Reviewer: [question] Is there a way to group or organize the different behavior sections on the page? Right now, it seems like the order is a bit arbitrary so I wonder if we can group some of these under parent headings (for ex, Reads and Writes as a parent over Read and Write Concern, Read Preference, Write Blocking, etc)?

Author: I like this idea. I created three sections: Considerations, Reads and Writes, and Destination Cluster Changes. What are your thoughts?

Reviewer: Is there something more specific than "Considerations" that we can use? I would assume that all of the sections on this page are considerations.

Reviewer: Alternatively, if there is a "miscellaneous" section of things that don't fit into other specific categories, I think we can put that section at the end instead of the beginning.

Author: Ok, I added "Cluster Types" and "Performance." Since "Capped Collections" was the only section left over, I moved it under "Destination Cluster Changes" because it does discuss temporary changes. I am not sure if this was the right call. Thoughts?

> Behavior

Reviewer: Since the title of this page is already "Behavior", I think we can remove this heading a bump ever other heading up (ie, making all the H3s into H2s and so on)

Author: Good idea, doing now!

--------

> Cluster Independence

Reviewer: I would suggest moving Cluster Independence, Configuration File, and Resilience into a subsection as well, if everything else is also in a subsection

connecting to the source and destination clusters. For more information,
see :ref:`connections-read-preference`. 

> Destination Cluster Changes
> ---------------------------

Reviewer: ```suggestion
Destination Cluster Considerations
-----------------------------------
```

Author: This makes sense, applying change.

time. However, if new writes are halted on the source cluster, the
source and destination clusters will eventually hold the same data.

Individual writes may be reordered or combined as they are relayed from
source to destination. For a given document, the total number of writes
may differ between source and destination.

Transactions may appear non-atomically on the destination cluster.
Retryable writes may not be retryable on the destination cluster.

Profiling
~~~~~~~~~

If profiling is enabled on a source database, MongoDB creates a special
collection named ``<db>.system.profile``. After synchronization is
complete, {+c2c-product-name+} will not drop the
``<db>.system.profile`` collection from the destination even if the
source database is dropped at a later time. The ``<db>.system.profile``
collection will not change the accuracy of user data on the
destination.

Views
~~~~~

If a database with views is dropped on the source, the destination may
show an empty ``system.views`` collection in that database. The empty
``system.views`` collection will not change the accuracy of user
data on the destination.

.. _c2c-capped-collections:

> Capped Collections

Reviewer: Since capped collections and views have information that has to do with both source and destination clusters, how about a "Collection Considerations" section?

Reviewer: Alternatively, "Capped Collections" can go in the collection considerations alone. I'm just hesitant to put it under Destination Cluster Changes since the info in this section has to do with both the source and the destination

Author: Discussed in slack -- creating "Cluster and Collection Types" heading.

connecting to the source and destination clusters. For more information,
see :ref:`connections-read-preference`. 

> Destination Cluster Considerations
> ----------------------------------

Reviewer: Since all of these sections are considerations (ie, Cluster and Collection Type Considerations, Performance Considerations, etc) I think we can remove "Considerations" here.
```suggestion
Destination Clusters
--------------------
```

Author: Makes sense, making change now!
